anonymize_test.py

```diff
@@ -0,0 +1,24 @@
+from anonymize import anonymize
+from utils import pdf_to_txt, txt_to_string
+
+
+
+def test_anonymize_name():
+    anonymize("test_example.pdf")
+    text=txt_to_string('anonymized_text.txt')
+    assert "John Doe" not in text and "John Doe" in pdf_to_txt("test_example.pdf")
+
+
+def test_anonymize_phone():
+    text=txt_to_string('anonymized_text.txt')
+    assert "07816093423" not in text and "07816093423" in pdf_to_txt("test_example.pdf")
+
+def test_anonymize_email():
+    text=txt_to_string('anonymized_text.txt')
+    #doesn't pass the test since in .com the last character "m" is contained in the anonymized file here in (text) just like this "<anonymized>m"
```

Comment:
it's surprising that it doesn't pass, there's probably a way to configure Presidio to make it detect email addresses.

Reply:
yes maybe we can precise but in the previous example all the email addresses were anonymized completely so here it's more about the conversion from pdf to txt maybe a space has been added or smth removed so the last "m" left alone

Reply:
oh ok, that's annoying for us but that's interesting. this issue actually comes from `PdfReader` that's is not the best library in python to read pdf files. here it the txt output by your current function:

```
Hello my name is John Doe , my phone number is 07816093423. Let me share with you my email 
address louisbond@example.co m and also my physical address  22 Rue Didouche Mourad Algiers, 
Algeria
```

there's a more used python library called [pdfplumber](https://github.com/jsvine/pdfplumber) (look at pdf stars). if you implement your function that way:

```python
import pdfplumber

def pdf_to_txt_2(filename):
    text = ""
    with pdfplumber.open(filename) as pdf:
        for page in pdf.pages:
            text += page.extract_text()
    return text
```

you have this result: 

```
Hello my name is John Doe, my phone number is 07816093423. Let me share with you my email
address louisbond@example.com and also my physical address 22 Rue Didouche Mourad Algiers,
Algeria
```

Reply:
tdlr: the library [pdfreader](https://github.com/maxpmaxp/pdfreader) parsed an email address with `name@example.co m`, while the library [pdfplumber](https://github.com/jsvine/pdfplumber) parsed it with `name@example.com`.